Wrong Client Authn Method at token endpoint throws NPE

![image](https://user-images.githubusercontent.com/3717101/210408617-a1a84669-78eb-49a6-9e73-e6833d94f76d.png)

If your client uses a different authn method at the token endpoint, this very unfriendly message is thrown by Auth Server.

```
Caused by: java.lang.NullPointerException
	at io.jans.as.server.service.AuthorizeService.permissionGranted(AuthorizeService.java:166) ~[classes/:?]
	at io.jans.as.server.service.AuthorizeService$Proxy$_$$_WeldClientProxy.permissionGranted(Unknown Source) ~[classes/:?]
	at io.jans.as.server.authorize.ws.rs.AuthorizeAction.permissionGranted(AuthorizeAction.java:809) ~[classes/:?]
	at io.jans.as.server.authorize.ws.rs.AuthorizeAction.permissionGranted(AuthorizeAction.java:804) ~[classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.apache.el.parser.AstValue.invoke(AstValue.java:252) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
	at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:266) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
	at org.jboss.weld.module.web.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) ~[weld-web-4.0.3.Final.jar:4.0.3.Final]
	at org.jboss.weld.module.web.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) ~[weld-web-4.0.3.Final.jar:4.0.3.Final]
	at com.sun.faces.facelets.el.TagMethodExpression.invoke(TagMethodExpression.java:70) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.application.MethodBindingMethodExpressionAdapter.invoke(MethodBindingMethodExpressionAdapter.java:67) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.application.ActionListenerImpl.getNavigationOutcome(ActionListenerImpl.java:75) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:64) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.component.UICommand.broadcast(UICommand.java:213) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:852) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.component.UIViewRoot.processApplication(UIViewRoot.java:1329) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.lifecycle.InvokeApplicationPhase.execute(InvokeApplicationPhase.java:56) ~[jakarta.faces-3.0.2.jar:3.0.2]
	... 44 more
```

hi @yuriyz ,
i checked logs it is same error in my log file as well,
please find log zip attached we well

Caused by: java.lang.NullPointerException
	at io.jans.as.server.service.AuthorizeService.permissionGranted(AuthorizeService.java:166) ~[classes/:?]
	at io.jans.as.server.service.AuthorizeService$Proxy$_$$_WeldClientProxy.permissionGranted(Unknown Source) ~[classes/:?]
	at io.jans.as.server.authorize.ws.rs.AuthorizeAction.permissionGranted(AuthorizeAction.java:809) ~[classes/:?]
	at io.jans.as.server.authorize.ws.rs.AuthorizeAction.permissionGranted(AuthorizeAction.java:804) ~[classes/:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:566) ~[?:?]
	at org.apache.el.parser.AstValue.invoke(AstValue.java:252) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
	at org.apache.el.MethodExpressionImpl.invoke(MethodExpressionImpl.java:266) ~[org.mortbay.jasper.apache-el-10.0.14.jar:10.0.14]
	at org.jboss.weld.module.web.util.el.ForwardingMethodExpression.invoke(ForwardingMethodExpression.java:40) ~[weld-web-4.0.3.Final.jar:4.0.3.Final]
	at org.jboss.weld.module.web.el.WeldMethodExpression.invoke(WeldMethodExpression.java:50) ~[weld-web-4.0.3.Final.jar:4.0.3.Final]
	at com.sun.faces.facelets.el.TagMethodExpression.invoke(TagMethodExpression.java:70) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.application.MethodBindingMethodExpressionAdapter.invoke(MethodBindingMethodExpressionAdapter.java:67) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.application.ActionListenerImpl.getNavigationOutcome(ActionListenerImpl.java:75) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.application.ActionListenerImpl.processAction(ActionListenerImpl.java:64) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.component.UICommand.broadcast(UICommand.java:213) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.component.UIViewRoot.broadcastEvents(UIViewRoot.java:852) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.component.UIViewRoot.processApplication(UIViewRoot.java:1329) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.lifecycle.InvokeApplicationPhase.execute(InvokeApplicationPhase.java:56) ~[jakarta.faces-3.0.2.jar:3.0.2]
	... 49 more
2023-01-04 08:22:09,458 ERROR [qtp952562199-15] b693654d-3bf4-4dca-a5f3-363a684f2599 [io.jans.as.server.exception.GlobalExceptionHandler] (GlobalExceptionHandler.java:75) - #{authorizeAction.permissionGranted}: java.lang.NullPointerException
jakarta.faces.FacesException: #{authorizeAction.permissionGranted}: java.lang.NullPointerException
	at com.sun.faces.lifecycle.InvokeApplicationPhase.execute(InvokeApplicationPhase.java:64) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.lifecycle.Phase.doPhase(Phase.java:72) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:159) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.webapp.FacesServlet.executeLifecyle(FacesServlet.java:681) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at jakarta.faces.webapp.FacesServlet.service(FacesServlet.java:437) ~[jakarta.faces-3.0.2.jar:3.0.2]
	at org.eclipse.jetty.servlet.ServletHolder$NotAsync.service(ServletHolder.java:1419) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:764) ~[?:?]
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1665) ~[?:?]
	at io.jans.as.server.audit.debug.ServletLoggingFilter.doFilter(ServletLoggingFilter.java:71) ~[classes/:?]




[authlogs.zip](https://github.com/JanssenProject/jans/files/10349660/authlogs.zip)

--- a/jans-auth-server/server/src/main/java/io/jans/as/server/service/AuthorizeService.java
+++ b/jans-auth-server/server/src/main/java/io/jans/as/server/service/AuthorizeService.java
@@ -151,13 +151,18 @@ public class AuthorizeService {
         try {
             final User user = sessionIdService.getUser(session);
             if (user == null) {
-                log.debug("Permission denied. Failed to find session user: userDn = " + session.getUserDn() + ".");
+                log.debug("Permission denied. Failed to find session user: userDn = {}", session.getUserDn());
                 permissionDenied(session);
                 return;
             }
 
             String clientId = session.getSessionAttributes().get(AuthorizeRequestParam.CLIENT_ID);
             final Client client = clientService.getClient(clientId);
+            if (client == null) {
+                log.debug("Permission denied. Failed to find client by id: {}", clientId);
+                permissionDenied(session);
+                return;
+            }
 
             String scope = session.getSessionAttributes().get(AuthorizeRequestParam.SCOPE);
             String responseType = session.getSessionAttributes().get(AuthorizeRequestParam.RESPONSE_TYPE);

